packages/utilities/src/guards/Rate Limiter/RateLimit.ts

```diff
@@ -100,19 +120,20 @@ export function RateLimit(
       memberId = arg?.member?.user?.id;
       guildId = arg?.guildId;
     }
-
     if (!memberId || !guildId) {
       return next();
     }
-
     let fromArray = getFromArray(memberId, guildId);
-    if (fromArray && message) {
-      return post(arg, message);
+    if (fromArray) {
+      fromArray.incrementCallCount();
+      if (fromArray.hasLimitReached()) {
+        return post(arg, message);
+      }
+      _timer.refresh(fromArray);
     } else {
-      fromArray = new TimeOutEntry(memberId, guildId);
+      fromArray = new TimeOutEntry(memberId, guildId, rateValue);
       _timer.add(fromArray);
     }
-
     return next();
   };
-}
```

Comment:
revert

Reply:
why? there should be a semicolon here

Reply:
why?

Reply:
oh you have used arrow function here, so use normal function instead.